drivers/dac/ad7293/ad7293.c

```diff
@@ -0,0 +1,532 @@
+		      unsigned int ch, uint16_t offset)
+{
+	unsigned int reg_wr;
+
+	switch (type) {
+	case AD7293_ADC_VINX:
+		reg_wr = AD7293_REG_VIN0_OFFSET;
+
+		break;
+	case AD7293_ADC_TSENSE:
+		reg_wr = AD7293_REG_TSENSE_INT_OFFSET;
+
+		break;
+	case AD7293_ADC_ISENSE:
+		reg_wr = AD7293_REG_ISENSE0_OFFSET;
+
+		break;
+	case AD7293_DAC:
+		reg_wr = AD7293_REG_UNI_VOUT0_OFFSET;
+
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return ad7293_spi_write(dev, reg_wr + ch, offset);
+}
+
+/**
+ * @brief Set the DAC output raw value.
+ * @param dev - The device structure.
+ * @param ch - the channel number.
+ * @param raw - the raw value to be written.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int ad7293_dac_write_raw(struct ad7293_dev *dev, unsigned int ch,
+			 uint16_t raw)
+{
+	int ret;
+
+	ret = ad7293_spi_update_bits(dev, AD7293_REG_DAC_EN, NO_OS_BIT(ch),
+				     NO_OS_BIT(ch));
+	if (ret)
+		return ret;
+
+	return ad7293_spi_write(dev, AD7293_REG_UNI_VOUT0 + ch,
+				no_os_field_prep(AD7293_REG_DATA_RAW_MSK, raw));
+}
+
+/**
+ * @brief Read raw value for specific channel and channel type.
+ * @param dev - The device structure.
+ * @param type - The channel type.
+ * @param ch - the channel number.
+ * @param raw - the raw value read.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int ad7293_ch_read_raw(struct ad7293_dev *dev, enum ad7293_ch_type type,
+		       unsigned int ch, uint16_t *raw)
+{
+	int ret;
+	unsigned int reg_wr, reg_rd, data_wr;
+
+	switch (type) {
+	case AD7293_ADC_VINX:
+		reg_wr = AD7293_REG_VINX_SEQ;
+		reg_rd = AD7293_REG_VIN0 + ch;
+		data_wr = NO_OS_BIT(ch);
+
+		break;
+	case AD7293_ADC_TSENSE:
+		reg_wr = AD7293_REG_ISENSEX_TSENSEX_SEQ;
+		reg_rd = AD7293_REG_TSENSE_INT + ch;
+		data_wr = NO_OS_BIT(ch);
+
+		break;
+	case AD7293_ADC_ISENSE:
+		reg_wr = AD7293_REG_ISENSEX_TSENSEX_SEQ;
+		reg_rd = AD7293_REG_ISENSE_0 + ch;
+		data_wr = NO_OS_BIT(ch) << 8;
+
+		break;
+	case AD7293_DAC:
+		reg_rd = AD7293_REG_UNI_VOUT0 + ch;
+
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (type != AD7293_DAC) {
+		if (type == AD7293_ADC_TSENSE) {
+			ret = ad7293_spi_write(dev, AD7293_REG_TSENSE_BG_EN,
+					       NO_OS_BIT(ch));
+			if (ret)
+				return ret;
+
+			no_os_mdelay(9);
+		} else if (type == AD7293_ADC_ISENSE) {
+			ret = ad7293_spi_write(dev, AD7293_REG_ISENSE_BG_EN,
+					       NO_OS_BIT(ch));
+			if (ret)
+				return ret;
+
+			no_os_mdelay(9);
+		}
+
+		ret = ad7293_spi_write(dev, reg_wr, data_wr);
+		if (ret)
+			return ret;
+
+		ret = ad7293_spi_write(dev, AD7293_REG_CONV_CMD, 0x82);
+		if (ret)
+			return ret;
+	}
+
+	ret = ad7293_spi_read(dev, reg_rd, raw);
+	if (ret)
+		return ret;
+
+	*raw = no_os_field_get(AD7293_REG_DATA_RAW_MSK, *raw);
+
+	return 0;
+}
+
+/**
+ * @brief Perform software reset.
+ * @param dev - The device structure.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int ad7293_soft_reset(struct ad7293_dev *dev)
```

Comment:
I see that the documentations specifies the following:
To issue a software reset write a specific value, 0x7293, to this
16-bit register and pull CS high. The user must write 0x0000 to
this register to clear it following the software reset. See Table 23.

I suppose that CS is pulled high by default, right?